test/browser/focus.test.js

```diff
@@ -377,6 +383,22 @@ describe('focus', () => {
 		validateFocus(input, 'remove sibling before 2');
 	});
 
+	it('should  maintain focus when removing element directly before input', () => {
```

Comment:
This test cases mimics what the repro in #3814 does. I can also add that specific repro if you'd like

Reply:
Fine be me the way you did it in the PR 👍